client/src/Header.js

```diff
@@ -44,6 +59,17 @@ function Header() {
             About
           </Link>
         </li>
+        {loading ? (
+          <></>
```

Comment:
Good work on setting up state to track loading, I think your implementation is correct...

Note that, since you're just rendering an empty fragment, there's no visible output to the user that the header is in the loading state. If you didn't have the loading state variable, it would still work and look the same, because you correctly set the default value for the categories state variable to be an empty array. So the category.map would not render anything either...